ckanext/stadtzhharvest/harvester.py

```diff
@@ -991,21 +1013,21 @@ def _get_immediate_subdirectories(self, directory):
 
     def _diff_path(self, package_id):
         today = datetime.date.today()
-        package_id = self._validate_package_id(package_id)
         if package_id:
             return os.path.join(
                 self.DIFF_PATH,
                 '%s-%s.html' % (str(today), package_id)
             )
 
-    def _create_notifications_for_deleted_datasets(self):
+    def _check_for_deleted_datasets(self, harvest_job):
         current_dirs = self._get_immediate_subdirectories(
             self.config['data_path']
         )
         current_datasets = [self._validate_package_id(d) for d in current_dirs]
         cached_datasets = self._get_immediate_subdirectories(
             os.path.join(self.DIFF_PATH, self.config['metadata_dir'])
         )
```

Comment:
I think we should change the way the `cached_datasets` are gathered. Currently the saved meta.xml's from the previous day are used to determine "old" datasets (as configured in metadata_dir on the harvest source).
This has a huge downside of relying on that directory, which could change anytime. I think it would be more stable and saner to rely on Solr. Simply get all datasets that are currently associated with the current harvest source at the beginning of a harvest job.